### PR TITLE
Delete entire database for Data when there is no user signed in

### PR DESCRIPTION
**Pull request by jaelim-ms**
* [ ] Has `CHANGELOG.md` been updated?
* [X] Are tests passing locally?
* [X] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [X] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Drop a database instead of dropping all tables. There is an error when it tries to drop SQLite reserved table.

### CODE REVIEW

File: AppCenter/AppCenter/Internals/Storage/MSDBStorage.h [hunk 1]
```diff
@@ -128,7 +128,7 @@ static NSString *const kMSSQLiteConstraintAutoincrement = @"AUTOINCREMENT";
  * Deletes all the tables within an existing database
```
Comment (dhei) [inline AppCenter/AppCenter/Internals/Storage/MSDBStorage.h]:
```suggestion
 * Deletes database.
```

File: AppCenterData/AppCenterData/Internal/LocalStore/MSDocumentStore.h [hunk 2]
```diff
@@ -74,7 +74,7 @@ NS_ASSUME_NONNULL_BEGIN
 /**
  * Delete all tables.
```
Comment (dhei) [inline AppCenterData/AppCenterData/Internal/LocalStore/MSDocumentStore.h]:
```suggestion
 * Delete database.
```

File: AppCenter/AppCenter/Internals/Storage/MSDBStorage.h [hunk 3]
```diff
@@ -125,10 +125,10 @@ static NSString *const kMSSQLiteConstraintAutoincrement = @"AUTOINCREMENT";
 - (BOOL)dropTable:(NSString *)tableName;
 
 /**
- * Deletes all the tables within an existing database
+  * Deletes database.
```
Comment (dhei) [inline AppCenter/AppCenter/Internals/Storage/MSDBStorage.h]:
```suggestion
 * Deletes database.
```

File: AppCenterData/AppCenterData/Internal/LocalStore/MSDocumentStore.h [hunk 4]
```diff
@@ -72,9 +72,9 @@ NS_ASSUME_NONNULL_BEGIN
 - (BOOL)deleteUserStorageWithAccountId:(NSString *)accountId;
 
 /**
- * Delete all tables.
+  * Delete database.
```
Comment (dhei) [inline AppCenterData/AppCenterData/Internal/LocalStore/MSDocumentStore.h]:
```suggestion
 * Delete database.
```

File: AppCenter/AppCenterTests/MSDBStorageTests.m [hunk 5]
```diff
@@ -385,7 +385,7 @@ - (void)testDroppAllTables {
   XCTAssertTrue([self tableExists:tableName1]);
   XCTAssertTrue([self.sut createTable:tableName2 columnsSchema:self.schema[kMSTestTableName]]);
   XCTAssertTrue([self tableExists:tableName2]);
-  [self.sut dropAllTables];
+  [self.sut dropDatabase];
 
   // Then
   XCTAssertFalse([self tableExists:tableName1]);
```
Comment (jwallra) [inline AppCenter/AppCenterTests/MSDBStorageTests.m]: The assertions here should change to verify that the database no longer exists

File: AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m [hunk 6]
```diff
@@ -439,7 +439,7 @@ - (void)testDeletionOfAllTables {
   XCTAssertTrue([self tableExists:tableName]);
 
   // When
-  [self.sut deleteAllTables];
+  [self.sut dropDatabase];
 
   // Then
   XCTAssertFalse([self tableExists:tableName]);
```
Comment (jwallra) [inline AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m]: Same here - assertions should change

File: AppCenter/AppCenterTests/MSDBStorageTests.m (hunk 5 above)
Comment (jaelim-ms) [reply AppCenter/AppCenterTests/MSDBStorageTests.m]: I keep this one and add file check assertion.

**Pull request merged**